### PR TITLE
#11 Add uri_param for audio_event_detections_clustering table

### PR DESCRIPTION
**Pull request by rassokhina-e**
ref: https://github.com/rfcx/arbimon-db/issues/11

there are 4 lambdas related to this table, 2 (large and small) for production namespace and 2 (large and small) for the staging namespace
Prod:
- https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/audio-event-detection-production-workerLarge-coX78heh6MKz?tab=code
- https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/audio-event-detection-production-worker-Wyo133bhN1Kr?tab=code

Stag:
- https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/audio-event-detection-staging-worker-mOfgj2SEOSat?tab=code
- https://us-east-1.console.aws.amazon.com/lambda/home?region=us-east-1#/functions/audio-event-detection-staging-conductor-qgRzLLtMqt45?tab=code

all these 4 lambdas just take to change the row from `'uri_image': image_uri+str(c)+'.png'` to `'uri_param': int(c)`

Once each lambda is updated, we need to create a new version of lambda to commit the changes 
Screenshot 2024-10-02 at 17 52 35
Screenshot 2024-10-02 at 17 52 29

### CODE REVIEW

File: init/056-add-uri-param-to-aed-clustering.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
+-- uri_image = 'audio_events/prod/detection/33604/png/3450166/3.png'
+
+ALTER TABLE `arbimon2`.`audio_event_detections_clustering`
+ADD COLUMN `uri_param` int DEFAULT NULL;
+
+UPDATE `arbimon2`.`audio_event_detections_clustering`
+SET uri_image = SELECT substring_index(substring_index(uri_image, '/', -1), '.', 1);
+
+UPDATE `arbimon2`.`audio_event_detections_clustering`
+SET uri_param = uri_image;
```
Comment (Tooseriuz) [inline init/056-add-uri-param-to-aed-clustering.sql]: cant we just do ?
```
UPDATE `arbimon2`.`audio_event_detections_clustering`
SET uri_param = SELECT substring_index(substring_index(uri_image, '/', -1), '.', 1);
```

**Comment by rassokhina-e** [comment]
done

**Review by Tooseriuz — APPROVED** [review]

File: init/056-add-uri-param-to-aed-clustering.sql [hunk 2]
```diff
@@ -0,0 +1,11 @@
+-- uri_image = 'audio_events/prod/detection/33604/png/3450166/3.png'
+
+ALTER TABLE `arbimon2`.`audio_event_detections_clustering`
+ADD COLUMN `uri_param` int DEFAULT NULL;
```
Comment (Tooseriuz) [inline init/056-add-uri-param-to-aed-clustering.sql]: I think `int` is too big. we can use smaller type with unsigned

File: init/056-add-uri-param-to-aed-clustering.sql [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+UPDATE `arbimon2`.`audio_event_detections_clustering`
+SET uri_param = CAST((SELECT substring_index(substring_index(uri_image, '/', -1), '.', 1)) AS SIGNED);
```
Comment (Tooseriuz) [inline init/056-add-uri-param-to-aed-clustering.sql]: better change the column type to `unsigned` and you can do `AS UNSIGNED`

**Pull request merged**